components/Ink/src/private/MDCSimpleInkLayer.m

```diff
@@ -0,0 +1,162 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCSimpleInkLayer.h"
+
+@implementation MDCSimpleInkLayer
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    [self commonMDCSimpleInkLayerInit];
+  }
+  return self;
+}
+
+- (instancetype)initWithLayer:(id)layer {
+  self = [super initWithLayer:layer];
+  if (self) {
+    [self commonMDCSimpleInkLayerInit];
+  }
+  return self;
+}
+
+- (void)commonMDCSimpleInkLayerInit {
+  _initialRadius = 0;
+  _finalRadius = 0;
+  _inkColor = [UIColor colorWithWhite:0 alpha:0.08f];
+  _startAnimActive = NO;
+  _endAnimDelay = 0;
+}
+
+- (void)setFrame:(CGRect)frame {
+  [super setFrame:frame];
+  self.initialRadius = (sqrt(frame.size.height * frame.size.height +
```

Comment:
Use MDCHypot.
self.initialRadius = MDCHypot(CGRectHeight(frame), CGRectWidth(frame)) / 2 * 06;

Reply:
@ianegordon super-uper-duper-nit:
self.initialRadius = **(CGFloat)**(MDCHypot(CGRectHeight(frame), CGRectWidth(frame)) / 2 * 0.6);

Reply:
@romoore If MDCHypot already returns a CGFloat, won't the 2 and 0.6 cast to CGFloat automatically?

Reply:
It's my understanding that specifying "0.6" as a double value will up-convert to `double` from `float`.  Since it's the final operation, the entire value will be a `double` when assigned back to `self.initialRadius`.

Reply:
Done.